src/model.ts

```diff
@@ -7,67 +7,123 @@ import { IContentFieldFilter, IContentSearchRequest } from '@esri/hub-search';
 import { PassThrough } from 'stream';
 import { PagingStream } from './paging-stream';
 import { getBatchedStreams } from './helpers/get-batched-streams';
-import { fetchSite, getHubApiUrl, getPortalApiUrl, hubApiRequest, RemoteServerError } from '@esri/hub-common';
+import { fetchSiteModel, getHubApiUrl, getPortalApiUrl, hubApiRequest, IModel, RemoteServerError } from '@esri/hub-common';
+
+const REQUIRED_FIELDS = [
+  'id', // used for the dataset landing page URL
+  'type', // used for the dataset landing page URL
+  'slug', // used for the dataset landing page URL
+  'access', // used for detecting proxied csv's
+  'size', // used for detecting proxied csv's
+  'licenseInfo', // required for license resolution
+  'structuredLicense', // required for license resolution
+];
+
+// additional fields due to dataset enrichment
+const ADDON_FIELDS = [
+  'hubLandingPage',
+  'accessUrlCSV',
+  'isLayer',
+  'accessUrlKML',
+  'accessUrlShapeFile',
+  'accessUrlWFS',
+  'accessUrlWMS',
+  'accessUrlGeoJSON',
+  'license'
+];
+
+type HubApiRequest = {
+  res?: {
+    locals: {
+      searchRequestBody?: IContentSearchRequest,
+      siteModel?: IModel,
+      arcgisPortal?: string
+      siteIdentifier?: string,
+    }
+  }
+  query?: {
+    limit?: number
+  }
+}
 
 export class HubApiModel {
 
-  async getStream (request: Request) {
-    const searchRequest: IContentSearchRequest = request.res.locals.searchRequest;
-    this.preprocessSearchRequest(searchRequest);
+  async getStream(request: Request) {
+   
+    const { res: { locals: { searchRequestBody, siteModel, arcgisPortal, siteIdentifier } }, query: { limit } }: HubApiRequest = request;
+    const hubSiteModel = siteModel || await this.fetchHubSiteModel(siteIdentifier, this.getRequestOptions(arcgisPortal));
+    this.preprocessSearchRequest(searchRequestBody);
 
-    if (searchRequest.options.fields) {
+    if (searchRequestBody.options.fields) {
+      searchRequestBody.options.fields = this.getValidHubApiFields(searchRequestBody.options.fields);
       const validFields: string[] = await hubApiRequest('/fields');
-      this.validateFields(searchRequest, validFields);
+      this.validateFields(searchRequestBody, validFields);
     }
 
     // Only fetch site if site is provided and either group or orgid is undefined
     if (
-      searchRequest.options.site &&
+      searchRequestBody.options.site &&
       (
-        !this.scopedFieldValueIsValid(searchRequest.filter.group) || 
-        !this.scopedFieldValueIsValid(searchRequest.filter.orgid)
+        !this.scopedFieldValueIsValid(searchRequestBody.filter.group) ||
+        !this.scopedFieldValueIsValid(searchRequestBody.filter.orgid)
```

Comment:
Convert the check inside this `if` to a well name function that captures what is said in the comment, then delete the comment, e.g.:
`shouldFetchSite(searchRequestBody)`

Reply:
extracted the check to a function`shouldFetchSite(searchRequestBody)`